### DISCUSSION TITLE
Help to convert <a> tag to <Button> in pagination component

### DISCUSSION THREAD

**Discussion by mallusrgreatv2** [shadcn-ui/ui#2465] [Q&A]
I have tried numerous ways. All the time, it keeps spitting out errors. Can someone do it for me?

**Comment by Alexjoshua14** [comment]
Here's a quick jab at it, I changed the pagination file to a client component to account for the client-side interactivity required for buttons (although this could be cleaned up by moving the button component into a separate file), I then created a new component to replace PaginationLink called PaginationButton and replaced uses of PaginationLink with PaginationButton. I also think that any component calling this new pagination variation might need to be a client component as well in order to pass onClick functions. Here's a link to my [altered version of pagination](https://github.com/Alexjoshua14/shadcn-ui-sandbox/blob/main/src/components/ui/pagination.tsx). Hope this helps!
